AK/ObjectBuffer.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace AK {
+
+template<class T>
+class ObjectBuffer {
```

Comment:
I'd call it `AlignedObjectBuffer`, `ObjectBuffer` doesn't tell anything about its specialty.